src/wp-includes/class-wp-term-query.php

```diff
@@ -742,23 +744,27 @@ public function get_terms() {
 			return $this->terms;
 		}
 
-		// $args can be anything. Only use the args defined in defaults to compute the key.
-		$key          = md5( serialize( wp_array_slice_assoc( $args, array_keys( $this->query_var_defaults ) ) ) . serialize( $taxonomies ) . $this->request );
-		$last_changed = wp_cache_get_last_changed( 'terms' );
-		$cache_key    = "get_terms:$key:$last_changed";
-		$cache        = wp_cache_get( $cache_key, 'terms' );
-		if ( false !== $cache ) {
-			if ( 'all' === $_fields || 'all_with_object_id' === $_fields ) {
-				$cache = $this->populate_terms( $cache );
-			}
+		if ( $args['cache_results'] ) {
+			// $args can be anything. Only use the args defined in defaults to compute the key.
+			$key          = md5( serialize( wp_array_slice_assoc( $args, array_keys( $this->query_var_defaults ) ) ) . serialize( $taxonomies ) . $this->request );
+			$last_changed = wp_cache_get_last_changed( 'terms' );
+			$cache_key    = "get_terms:$key:$last_changed";
+			$cache        = wp_cache_get( $cache_key, 'terms' );
```

Comment:
I am completely unsure as to whether this line should be conditional on the `cache_results` parameter.

Technically the code bypasses an existing cache when the parameter is intended to prevent priming or refreshing the cache.

The _practical_ question is: do developers want to bypass a previously primed cached  if they set the new parameter to `true`?

🤷🤷🤷🤷🐰🕳️🤷🤷

Reply:
There is no point of adding add a cache here. Cache key is generated using $args, including `cache_results`. The second time this functoni is called, with `cache_results` => false, even if the cache is set, then it always be ignored. What is the point of setting it if it is never used?

Reply:
I'm specifically considering the getter, `wp_cache_get()`, rather than the setter. 

If the query matches a previously cached result, should `WP_Term_Query` use that even if `cache_results === false`?